Bayesian Linear Regression/ ANOVAs: updates for posterior summary with matched models

vandenman opened the pull request:
Fixes https://github.com/jasp-stats/INTERNAL-jasp/issues/795
Fixes https://github.com/jasp-stats/jasp-issues/issues/852
Fixes https://github.com/jasp-stats/INTERNAL-jasp/issues/914

The screenshot below shows the new posterior effects table. The table is ordered so that the first columns relate to variable selection and the last columns relate to the parameter estimates.

In addition, the posterior exclusion probabilities are now shown when `Matched models` is ticked. This allows people to compute the inclusion Bayes factor from this table, without having to compute the exclusion probabilities themselves from the other table (e.g., for `contcor1`, `(0.207 / 0.690) / (.222 / .444) = 0.600`.

![image](https://user-images.githubusercontent.com/21319932/89779645-25e5d380-db10-11ea-9e03-c1836364f75d.png)


The changes in the error handling function of the ANOVAs silence the unit tests of the repeated measures ANOVAs (https://travis-ci.org/github/vandenman/jasp-desktop/builds/716464613, line 1715 `Error in .subset(x, j) : invalid subscript type 'list'` compared to line 1715 of https://travis-ci.org/github/vandenman/jasp-desktop/builds/716516437 (this PR)).

vandenman committed:
update matched models with new order and post/prior exclusion probabilities

vandenman committed:
update unit-tests

vandenman committed:
fix and silence anova error check

vandenman committed:
remake tests with correct BAS version (1.5.4)

vandenman committed:
cleanup

AlexanderLyNL reviewed (approved):
Looks good to me. The reason why `P(exclude | data)` is not in there for across all models is because that's just `1-P(include | data)` isn't it? The fact that the column `P(exclude | data)` "all of a sudden" appears might scare the user.

AlexanderLyNL commented:
Including `P(exclude | data)` to the table for both cases might be illustrative I think, even though it's a bit redundant for across all models. Either way is fine for me, but having `P(exclude | data)` in the table by default makes it easier to show what across `Matched models` does.

vandenman commented:
> The reason why P(exclude | data) is not in there for across all models is because that's just 1-P(include | data) isn't it?

Yes.

> Either way is fine for me, but having P(exclude | data) in the table by default makes it easier to show what across Matched models does.

I don't have very strong feelings about that either, just let me know what you prefer and I'll change it to that.

AlexanderLyNL commented:
Let's have `P(exclude | data)` in by default. Thanks for improving this analysis by the way, it looks really great.

vandenman committed:
always show exclusion probabilities

vandenman committed:
updated unit tests for the 2 new columns

vandenman commented:
Exclusion probabilities are now always shown.

vandenman committed:
fix test value

it looks like Travis can actually compute the BF whereas my machine
returns Inf.

boutinb commented:
@TimKDJ: can we merge tis PR?

TimKDJ commented on JASP-Tests/R/tests/testthat/test-anovarepeatedmeasuresbayesian.R:
```diff
@@ -90,7 +90,7 @@ test_that("Effects table results match", {
 
   refTables <- list(
-    allModels = list("<unicode>", "RM_FACTOR_1", 0.4, 0, 0.6, 1, 2.08122183080571,
+    allModels = list(1000799917193443, "RM_FACTOR_1", 0.4, 0, 0.6, 1, 2.08122183080571,
```
Maybe you could use less extreme data so this will pass on different operating systems?

vandenman replied:
I didn't change the input data for this test, but I will now. If you don't mind I'll merge the test for the effects table and those for the main table since the two are independent (and the unit tests timed out a couple of times already).

vandenman committed:
rewrite bayesian repeated measures unit test

vandenman committed:
make the RM factor with non-replicable inclusion BF a nuisance variable

TimKDJ commented:
Can you make a PR to the module too?

vandenman commented:
here are the PRs to the modules:

https://github.com/jasp-stats/jaspRegression/pull/7
https://github.com/jasp-stats/jaspAnova/pull/8

these do not contain the changes to the unit tests, because those repos don't have any tests yet ¯\\_(ツ)_/¯
